assets/src/components/v2/train_crowding.tsx

```diff
@@ -0,0 +1,163 @@
+import CarSomeCrowdingMiddle from "../../../static/images/svgr_bundled/train_crowding/Car-SomeCrowding-Middle.svg";
+import CarCrowdedMiddle from "../../../static/images/svgr_bundled/train_crowding/Car-Crowded-Middle.svg";
+import CarNoDataMiddle from "../../../static/images/svgr_bundled/train_crowding/Car-NoData-Middle.svg";
+import CarClosedMiddle from "../../../static/images/svgr_bundled/train_crowding/Car-Closed-Middle.svg";
+
+import CarNotCrowdedRight from "../../../static/images/svgr_bundled/train_crowding/Car-NotCrowded-Right.svg";
+import CarSomeCrowdingRight from "../../../static/images/svgr_bundled/train_crowding/Car-SomeCrowding-Right.svg";
+import CarCrowdedRight from "../../../static/images/svgr_bundled/train_crowding/Car-Crowded-Right.svg";
+import CarNoDataRight from "../../../static/images/svgr_bundled/train_crowding/Car-NoData-Right.svg";
+import CarClosedRight from "../../../static/images/svgr_bundled/train_crowding/Car-Closed-Right.svg";
+
+interface Props {
+  arrivalTime: string,
+  crowding: string[],
+  destination: string,
+  front_car_direction: string,
+  now: string,
+  platform_position: number
+}
+
+const lookupCarComponent = (occupancyStatus: string, frontCarDirection: string | boolean) => {
+  if (frontCarDirection) {
+    if (occupancyStatus == "not_crowded") return frontCarDirection == "left" ? CarNotCrowdedLeft : CarNotCrowdedRight
+    else if (occupancyStatus == "some_crowding") return frontCarDirection == "left" ? CarSomeCrowdingLeft : CarSomeCrowdingRight
+    else if (occupancyStatus == "crowded") return frontCarDirection == "left" ? CarCrowdedLeft : CarCrowdedRight
+    else if (occupancyStatus == "closed") return frontCarDirection == "left" ? CarClosedLeft : CarClosedRight
+    else return frontCarDirection == "left" ? CarNoDataLeft : CarNoDataRight
+  } else {
+    if (occupancyStatus == "not_crowded") return CarNotCrowdedMiddle
+    else if (occupancyStatus == "some_crowding") return CarSomeCrowdingMiddle
+    else if (occupancyStatus == "crowded") return CarCrowdedMiddle
+    else if (occupancyStatus == "closed") return CarClosedMiddle
+    else return CarNoDataMiddle
+  }
+}
+
+const TrainCrowding: React.ComponentType<Props> = ({
+      crowding,
+      destination,
+      platform_position,
+      front_car_direction,
+      now
+    }) => {
+
+  // If the front car direction is right, the crowding array needs to be reversed
+  // so the last car is rendered on the left.
+  const trains = crowding.map((car, i) => {
+    const CarComponent = lookupCarComponent(car, i == 0 && front_car_direction)
+    return <CarComponent key={i} className="crowding-widget__train-car" />
+  })
+
+  const trainSequence = front_car_direction == "left" ? trains : ([] as ReactElement[]).concat(trains).reverse()
+
+  // If arrow is between screens, scoot the arrow to the next slot on the right
+  const arrowSlot = [1, 9, 17].includes(platform_position) ?
```

Comment:
Do we have documentation anywhere on what range of values `platform_position` can have? I checked on the backend and we set it straight from config (which makes sense) and it's not documented either in the config module or the widget instance module.

It would be helpful to add a comment somewhere about that.
If you think it would make more sense in the backend code I can add it in my PR to keep this one frontend-only—just let me know what values it can have.

Reply:
Good call! I think that'd make sense on the backend, where it pulls from the config. `platform_position` can be a positive number 1-25, inclusive